shell/common/rasterizer.cc

```diff
@@ -263,8 +301,9 @@ static sk_sp<SkData> ScreenshotLayerTreeAsImage(
   SkMatrix root_surface_transformation;
   root_surface_transformation.reset();
 
-  auto frame = compositor_context.AcquireFrame(
-      surface_context, canvas, nullptr, root_surface_transformation, false);
+  auto frame = compositor_context.AcquireFrame(surface_context, canvas, nullptr,
```

Comment:
@cyanglaz is just adding screenshot support, would it have any unwanted side affects to pass in the thread merger?

Reply:
My prototype doesn't rely on Skia to take screenshot, rather uses completely iOS apis. So the plan is to somehow check if there are embedded_views and if so, use the screenshot directly from the API that is provided from embedded views (iOS's PlatformViewController).

I am not too familiar with the thread merging but my guess is it won't affect the current behavior.

Need @iskakaushik' confirm or have a offline discussion about it.

Reply:
AFAIK for @cyanglaz 's screenshot change, the screenshot needs to happen on the platform thread since it operates on `CALayers`. It would make sense to keep the threads merged during the screenshot duration as its a rather rare operation.

I can help make that change.